### PR TITLE
Prepare for 1.24.1 Release 

### PR DESCRIPTION
**Pull request by koooosh**
**Issue number:**

#4235

**Description of changes:**

Update CHANGELOG.md, Twoliter.toml, and Release.toml for 1.24.1

**Testing done:**

N/A

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by larvacea — APPROVED** [review]

**Review by ginglis13 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
+# v1.24.1 (2024-10-04)
+
+## Release Highlights
+* Update ecs-agent to 1.86.3 [bottlerocket-core-kit#168](https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/168)
+* Update kmod-6.1-neuron to 2.18.12.0 [bottlerocket-core-kit#170](https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/170)
+* Update kernels to 5.10.226 and 5.15.167 [bottlerocket-core-kit#177](https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/177)
```
Comment (yeazelm) [inline CHANGELOG.md]: These don't really meet the bar for a highlight in my opinion. We should aim to provide highlights for features, bugfixes that have issues related to them, etc.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,23 @@
+# v1.24.1 (2024-10-04)
+
+## Release Highlights
+* Update ecs-agent to 1.86.3 [bottlerocket-core-kit#168](https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/168)
```
Comment (yeazelm) [inline CHANGELOG.md]: https://github.com/bottlerocket-os/bottlerocket/issues/4186 can we call out it closes this issue?

**Review by vyaghras — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (koooosh) [reply CHANGELOG.md]: Gotcha, will remove these 2 lines

**Comment by koooosh** [comment]
^ makes updates to the CHANGELOG.md

**Comment by koooosh** [comment]
Cancelling CI workflows since they previously already passed, and update only touches CHANGELOG

**Review by yeazelm — APPROVED** [review]
LGTM!

**Pull request merged**